src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -0,0 +1,683 @@
+			IL2087,
+			new LocalizableResourceString (nameof (Resources.DynamicallyAccessedMembersAnnotationsDoNotMatch),
+				Resources.ResourceManager, typeof (Resources)),
+			new LocalizableResourceString (nameof (Resources.IL2087),
+				Resources.ResourceManager, typeof (Resources)),
+			DiagnosticCategory.Trimming,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true);
+
+		public const string IL2088 = "IL2088";
+		static readonly DiagnosticDescriptor s_IL2088 = new (
+			IL2088,
+			new LocalizableResourceString (nameof (Resources.DynamicallyAccessedMembersAnnotationsDoNotMatch),
+				Resources.ResourceManager, typeof (Resources)),
+			new LocalizableResourceString (nameof (Resources.IL2088),
+				Resources.ResourceManager, typeof (Resources)),
+			DiagnosticCategory.Trimming,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true);
+
+		public const string IL2089 = "IL2089";
+		static readonly DiagnosticDescriptor s_IL2089 = new (
+			IL2089,
+			new LocalizableResourceString (nameof (Resources.DynamicallyAccessedMembersAnnotationsDoNotMatch),
+				Resources.ResourceManager, typeof (Resources)),
+			new LocalizableResourceString (nameof (Resources.IL2089),
+				Resources.ResourceManager, typeof (Resources)),
+			DiagnosticCategory.Trimming,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true);
+
+		public const string IL2090 = "IL2090";
+		static readonly DiagnosticDescriptor s_IL2090 = new (
+			IL2090,
+			new LocalizableResourceString (nameof (Resources.DynamicallyAccessedMembersAnnotationsDoNotMatch),
+				Resources.ResourceManager, typeof (Resources)),
+			new LocalizableResourceString (nameof (Resources.IL2090),
+				Resources.ResourceManager, typeof (Resources)),
+			DiagnosticCategory.Trimming,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true);
+
+		public const string IL2091 = "IL2091";
+		static readonly DiagnosticDescriptor s_IL2091 = new (
+			IL2091,
+			new LocalizableResourceString (nameof (Resources.DynamicallyAccessedMembersAnnotationsDoNotMatch),
+				Resources.ResourceManager, typeof (Resources)),
+			new LocalizableResourceString (nameof (Resources.IL2091),
+				Resources.ResourceManager, typeof (Resources)),
+			DiagnosticCategory.Trimming,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create (s_IL2067, s_IL2068, s_IL2069,
+			s_IL2070, s_IL2071, s_IL2072, s_IL2073, s_IL2074, s_IL2075, s_IL2076, s_IL2077, s_IL2078, s_IL2079,
+			s_IL2080, s_IL2081, s_IL2082, s_IL2083, s_IL2084, s_IL2085, s_IL2086, s_IL2087, s_IL2088, s_IL2089,
+			s_IL2090, s_IL2091);
+		#endregion
+
+		public override void Initialize (AnalysisContext context)
+		{
+			context.EnableConcurrentExecution ();
+			context.ConfigureGeneratedCodeAnalysis (GeneratedCodeAnalysisFlags.ReportDiagnostics);
+			context.RegisterSyntaxNodeAction (DynamicallyAccessedMembersAnalyze,
```

Comment:
I think an IOperation context makes more sense here -- syntax seems hard because you may have a lot of different syntaxes for the same semantic operation.